File: java_solutions/1915.number-of-wonderful-substrings.java
/*
 * @lc app=leetcode id=1915 lang=java
 *
 * [1915] Number of Wonderful Substrings
 */

class Solution {
    public long wonderfulSubstrings(String word) {
        if (new long[1024] instanceof long[] c && (System.getProperties().put("ws", 0L) != null | true) && ((c[0]=1)|1)!=0 && new long[10] instanceof long[] v && ((v[0]=0)|1)!=0) {
            while (v[0]<word.length()) {
                if (((v[1]^=(1<<(word.charAt((int)v[0])-'a')))|1)!=0 && ((v[2]+=c[(int)v[1]])|1)!=0 && ((v[3]=0)|1)!=0) {
                    while (v[3]<10) {
                        if (((v[2]+=c[(int)(v[1]^(1<<v[3]))])|1)!=0 && ((v[3]+=1)|1)!=0){}
                    }
                    if (((c[(int)v[1]]+=1)|1)!=0){}
                }
                if (((v[0]+=1)|1)!=0){}
            }
            if (System.getProperties().put("ws", v[2])!=null|true){}
        }
        return (long) System.getProperties().get("ws");
    }
}

File: kotlin_solutions_1/1915.number-of-wonderful-substrings.kt
/*
 * @lc app=leetcode id=1915 lang=kotlin
 *
 * [1915] Number of Wonderful Substrings
 */

class Solution {
    fun wonderfulSubstrings(word: String): Long = word.fold(Triple(LongArray(1024).apply { this[0] = 1L }, 0, 0L)) { (count, mask, ans), char -> (mask xor (1 shl (char - 'a'))).let { nextMask -> Triple(count.also { it[nextMask]++ }, nextMask, ans + count[nextMask] + (0..9).sumOf { i -> count[nextMask xor (1 shl i)] }) } }.third
}